x/cellarfees/keeper/test_common.go

```diff
@@ -0,0 +1,635 @@
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/cosmos/cosmos-sdk/x/capability"
+	"github.com/cosmos/cosmos-sdk/x/crisis"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	distrclient "github.com/cosmos/cosmos-sdk/x/distribution/client"
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/cosmos/cosmos-sdk/x/evidence"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/cosmos/cosmos-sdk/x/mint"
+	"github.com/cosmos/cosmos-sdk/x/params"
+	paramsclient "github.com/cosmos/cosmos-sdk/x/params/client"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	paramsproposal "github.com/cosmos/cosmos-sdk/x/params/types/proposal"
+	"github.com/cosmos/cosmos-sdk/x/slashing"
+	slashingkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/cosmos/cosmos-sdk/x/upgrade"
+	upgradeclient "github.com/cosmos/cosmos-sdk/x/upgrade/client"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/peggyjv/gravity-bridge/module/v2/x/gravity"
+	gravitykeeper "github.com/peggyjv/gravity-bridge/module/v2/x/gravity/keeper"
+	gravitytypes "github.com/peggyjv/gravity-bridge/module/v2/x/gravity/types"
+	"github.com/peggyjv/sommelier/v4/x/cellarfees/types"
+	corkkeeper "github.com/peggyjv/sommelier/v4/x/cork/keeper"
+	corktypes "github.com/peggyjv/sommelier/v4/x/cork/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+)
+
+var (
+	// ModuleBasics is a mock module basic manager for testing
+	ModuleBasics = module.NewBasicManager(
+		auth.AppModuleBasic{},
+		genutil.AppModuleBasic{},
+		bank.AppModuleBasic{},
+		capability.AppModuleBasic{},
+		staking.AppModuleBasic{},
+		mint.AppModuleBasic{},
+		distribution.AppModuleBasic{},
+		gov.NewAppModuleBasic(
+			paramsclient.ProposalHandler, distrclient.ProposalHandler, upgradeclient.ProposalHandler, upgradeclient.CancelProposalHandler,
+		),
+		params.AppModuleBasic{},
+		crisis.AppModuleBasic{},
+		slashing.AppModuleBasic{},
+		upgrade.AppModuleBasic{},
+		evidence.AppModuleBasic{},
+		vesting.AppModuleBasic{},
+		gravity.AppModuleBasic{},
+	)
+)
+
+var (
```

Comment:
This is more of a high level comment open to larger discussion, but perhaps a cleaner alternative to these large test_common.go files we have would be to mock out any cross module interactions we have to isolate modules for unit testing.

For reference the core cosmos modules do this by using gmock to mock out the expected keepers: 
https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/testutil/expected_keepers_mocks.go

Then they create some test helpers to mock the methods they need: 
https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/keeper/keeper_test.go#L139

And finally setup their tests: 
https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/keeper/keeper_test.go#L87

And use the mock methods ad hoc: 
https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/keeper/keeper_test.go#L307

For further reference this is how auction module unit testing was [approached](https://github.com/PeggyJV/sommelier/pull/180) as well as how Unique attempted to clean up the cork tests in an [older pr](https://github.com/PeggyJV/sommelier/pull/152).

This would indeed be a fairly significant change to how we approach unit tests, and perhaps is more of a stylistic/meta question as to how we as a team want to unit test our code vs integration test it (and perhaps further narrow/define those terms)

Thoughts?